dts/arm/st/h7/stm32h757.dtsi

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2024 Grinn sp. z o.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/h7/stm32h747.dtsi>
+
+/ {
+	soc {
+		compatible = "st,stm32h757", "st,stm32h7", "simple-bus";
+
+		cryp: cryp@48021000 {
+			compatible = "st,stm32-cryp";
+			reg = <0x48021000 0x400>;
+			clocks = <&rcc STM32_CLOCK(AHB2, 4U)>;
+			interrupts = <79 0>;
+			status = "disabled";
+		};
+
+		hash: cryp@48021400 {
+			compatible = "st,stm32-cryp";
+			reg = <0x48021400 0x400>;
+			clocks = <&rcc STM32_CLOCK(AHB2, 5U)>;
+			interrupts = <80 0>;
+			status = "disabled";
```

Comment:
hash and cryp are 2 different IPs. To my knowledge, this is the first time `st,stm32-cryp` compatible is applied to HASH IP. Also Crypto STM32 driver is not multi instance yet, so using both at a time won't work.

Do you have a plan for this or was it a wild guess ?
If you don't have specific plans, I'd propose to drop and hash support could be added later.

Reply:
It is based on existing SoC definition for STM32H755Xi: [stm32h755.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/st/h7/stm32h755.dtsi). Both SoCs share the same crypto peripherals, so it was added here in the same way.

I have no plans regarding hash peripheral. Should we also remove the same node from the other SoC?

Reply:
@erwango I've updated the PR and removed the crypto node.